apollo-ios-codegen/Sources/GraphQLCompiler/JavaScript/src/__tests__/deferDirectiveTests.ts

```diff
@@ -461,5 +464,162 @@ describe("given schema with valid defer directive", () => {
+  `;
+
+  // Directive Definition Tests
+
+  describe("automatically adds a defer directive", () => {
+    const documentString: string = `
+    query BooksQuery {
+      books {
+        title
+        ... on Book @defer(label: "deferredBookAuthor") {
+          author
+        }
+      }
+    }
+    `;
+
+    const document: DocumentNode = parseOperationDocument(
+      new Source(documentString, "Test Query", { line: 1, column: 1 })
+    );
+
+    it("ensure the introspection JSON is correct for test", () => {
```

Comment:
Introspection JSON is difficult to parse and read so I've added tests like this one to ensure that the JSON doesn't get changed in a way that would invalidate what the test is meant to be checking for.